### PR TITLE
CSS Infrastructure: Use grid to fix layout issues

### PR DESCRIPTION
**Pull request by margaree**
The main grid looks like:
```
Sidenav Content
Sidenav Footer
```

The content grid is the following:
```
Left Content Right
Left Content Right
...
```
This was because we want to be able to have the entire width of the content with a specified background colour with a minimum left/right padding of `2.7rem` but having auto size on the left/right overall.

### CODE REVIEW

File: .eleventy.cjs [hunk 1]
```diff
@@ -45,7 +45,7 @@ module.exports = function(eleventyConfig) {
 			return '';
 		}
 		const branchName = defaultBranch ?? 'main';
-		return `<d2l-icon icon="tier1:edit"></d2l-icon><div class="d2l-edit-component-page">Suggest an <d2l-link href="${`${repo}edit/${branchName}/${componentPath}`}">edit</d2l-link> for this page - contributions are welcome!</div>`;
+		return `<div class="d2l-edit-component-page"><d2l-icon icon="tier1:edit"></d2l-icon>Suggest an <d2l-link href="${`${repo}edit/${branchName}/${componentPath}`}">edit</d2l-link> for this page - contributions are welcome!</div>`;
```
Comment (margaree) [inline .eleventy.cjs:48]: This is to improve the wrapping behaviour of this content

File: .eleventy.cjs [hunk 2]
```diff
@@ -145,8 +145,22 @@ module.exports = function(eleventyConfig) {
...
+		 * h1 & h2:
+		 * 		div d2l-daylight-section-wrapper
+		 * 		div d2l-daylight-container-left /div
+		 * 		div d2l-daylight-section-wrapper-inner-content
+		 * h2:
+		 * 		</div> (end d2l-daylight-section-wrapper-inner-content)
+		 * 		div d2l-daylight-container-right /div
+		 * 		</div> (end previous d2l-daylight-section-wrapper)
+		 */
+		if (tag === 'h2') result += '</div><div class="d2l-daylight-container-right"></div></div>';
```
Comment (margaree) [inline .eleventy.cjs:161]: This piece is complex - it's to do with creating the content grid for each section, where sections start with either an h1 or h2, and the grid is
```
LeftSection Content RightSection
```

File: pages/_includes/layouts/component-issue.njk [hunk 3]
```diff
@@ -13,32 +13,36 @@ permalink: "components/{{fileName}}/"
...
-			<div class="d2l-daylight-h1-container">
-				<div>{{ title }}<img class="d2l-heading-img" src="/img/labs-icon.svg" alt="labs icon" /></div>
-	{% else %}
-		<h1 class="d2l-heading-1 d2l-daylight-request">
-			<div class="d2l-daylight-h1-container">
-				<div>{{ title }}<img class="d2l-heading-img" src="/img/request-icon.svg" alt="lightbulb icon" /></div>
-	{%- endif -%}
-	<d2l-daylight-component-status dev-status="{{ development }}" stability-status="{{ stability }}"></d2l-daylight-component-status></div>
-	</h1>
+<div class="d2l-daylight-section-wrapper">
```
Comment (margaree) [inline pages/_includes/layouts/component-issue.njk:16]: The content is the same, just modified to have the same grid layout as component and simple-content layouts

File: pages/_includes/css/style.css [hunk 4]
```diff
@@ -83,9 +63,42 @@ div.d2l-daylight-content {
...
+}
 .d2l-daylight-content h2.d2l-heading-2:focus-visible {
 	outline: none;
 }
+
+.d2l-daylight-section-wrapper {
+	display: grid;
+	grid-template-columns: auto minmax(284px, 1253px) auto;
+	padding-left: 2.7rem;
+	padding-right: 2.7rem;
```
Comment (dlockhart) [inline pages/_includes/css/style.css:77]: Not sure if it'll be helpful, but there's also `column-gap: 2.7rem` and `row-gap` which can help with the gutters between the cells.

Comment (margaree) [reply pages/_includes/css/style.css:77]: Ah nice yeah that can definitely replace this padding and the footer top margin

**Review by dlockhart — APPROVED** [review]

**Pull request merged**